src/Components/Components/src/ParameterViewBuilder.cs

```diff
@@ -44,10 +44,6 @@ public void Add(string name, object? value)
         /// <returns>The <see cref="ParameterView" />.</returns>
         public ParameterView ToParameterView()
         {
-            // Since this is internal, we should expect the usage to always be correct,
-            // i.e. the given count should match the number of 'Add' calls.
-            Debug.Assert(_frames[0].ElementSubtreeLengthField == _frames.Length);
```

Comment:
This check was unhelpful because the capacity needs to be understood as a *max* capacity, not a guarantee that it will be filled entirely. I realised that if the JS-side code tries to supply `undefined`, because of JSON serialization conventions, that parameter would be omitted, so it's possible for the number of supplied parameters to be less than the total count.